### PR TITLE
fix!: update to Lux & Boltz 1.0

### LINKED ISSUE #784: reactivate coverage
https://github.com/SciML/DiffEqFlux.jl/blob/01173be0cbc56bc862336d26d85cb7a0e601dfca/.github/workflows/CI.yml#L41

### PR DESCRIPTION
**Pull request by avik-pal**
Needs https://github.com/LuxDL/Boltz.jl/pull/57.

Needs https://github.com/JuliaML/MLDatasets.jl/pull/243

fixes #784

### CODE REVIEW

File: docs/src/examples/hamiltonian_nn.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ res = Optimization.solve(opt_prob, opt, dataloader; callback)
 
 ps_trained = res.u
 
-model = NeuralHamiltonianDE(
+model = NeuralODE(
```
Comment (ChrisRackauckas) [inline docs/src/examples/hamiltonian_nn.md:60]: Why was this changed away from Hamiltonian?

Comment (avik-pal) [reply docs/src/examples/hamiltonian_nn.md:60]: https://github.com/SciML/DiffEqFlux.jl/pull/945/files/42fe746f412813b1f0bbdcbac1bc99e6c2bbe5fb#diff-8af080be8793cf24a571c91750a7b2432e3820c5c45f68f3f1d13937d044bfa9L36 constructs the hamiltonian. In the last release this function was mostly a no-op but I forgot to change it

File: docs/src/examples/hamiltonian_nn.md [hunk 2]
```diff
@@ -95,12 +95,12 @@ dataloader = ncycle(
     NEPOCHS)
 ```
 
-### Training the HamiltonianNN
+### Training the
```
Comment (avik-pal) [inline docs/src/examples/hamiltonian_nn.md:98]:
```suggestion
### Training the HamiltonianNN
```

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Getting errors in the example https://buildkite.com/julialang/diffeqflux-dot-jl/builds/1682#0191fd62-73aa-412e-ad59-f3d71bf8a272/1796-2850
